### ISSUE TITLE
`override` decorator doesn't report problems when paired with `cached_property` decorator

### ISSUE THREAD

**Issue by pappnu** [microsoft/pyright#11130]
Pyright doesn't report a problem for a method decorated with `cached_property` and `override` in a class even if no such method exists in the base class. Is this intended behaviour?

```python
from functools import cached_property
from typing import override


class A:
    pass
    
class B(A):
    @cached_property
    @override
    def should_override(self) -> int: # Should report a problem
        return 0
```

Pylance 2025.9.1  
Pyright 1.1.406

**Comment by erictraut** [comment]
The `@cached_property` decorator transforms the function into a descriptor object, so the `@override` becomes irrelevant in this case.

I don't consider this a bug, but it could be a feature request to add a check for this case. I don't think this is a significant cause for bugs though, so it's low priority.

**Comment by rsdenijs** [comment]
FWIW, I just ran into this exact issue.
